.github/workflows/deploy-portainer.yaml

```
@@ -0,0 +1,160 @@
+name: Deploy portainer
+
+on:
+    workflow_call:
+        inputs:
+            registry:
+                required: true
+                type: string
+                default: "ghcr.io"
+            image_name:
+                required: true
+                type: string
+            docker_component:
+                required: true
+                type: string
+            deploy_environment:
+                required: true
+                type: string
+            deploy_port:
+                required: true
+                type: number
+            volumes:
+                required: true
+                type: string
+            port_bindings:
+                required: true
+                type: number
+            exposed_ports:
+                required: true
+                type: number
+        secrets:
+            PORTAINER_URL:
+                required: true
+            PORTAINER_API_KEY:
+                required: true
+            PORTAINER_REGISTRY_AUTH:
+                required: true
+            GH_TOKEN:
+                required: true
+concurrency:
+    group: deploy-portainer-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    deploy-docker:
+        name: Deploy docker image
+        runs-on: self-hosted
+        timeout-minutes: 15
+        steps:
+            - name: Login to Github docker registry
+              uses: docker/login-action@v1
+              with:
+                  registry: ${{ inputs.registry }}
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GH_TOKEN }}
+            - name: Pull image
+              run: |
+                  curl --insecure --request POST '${{ secrets.PORTAINER_URL }}/images/create?fromImage=${{ inputs.registry }}/${{ inputs.image_name }}&tag=latest' \
```

to make it even more generic we could pass the tag as an input too
```suggestion
                  curl --insecure --request POST '${{ secrets.PORTAINER_URL }}/images/create?fromImage=${{ inputs.registry }}/${{ inputs.image_name }}&tag=${{ inputs.tag}}' \
```